src/core/request/cache.js

```diff
@@ -48,6 +49,18 @@ export class CacheRequest extends Request {
     this._aggregation = aggregation;
   }
 
+  get collection() {
+    if (isDefined(this.tag)) {
+      return  this._collection + this.tag;
```

Comment:
I think it would be better to separate the collection name and tag by some symbol (or a number of symbols). Preferably a symbol, which cannot be used in a collection name, but can be used in a database name. It would give us some flexibility - we would be able to separate the name from the tag, which can be useful.

Reply:
I am asking @heyzooi what they did for iOS. We will do the same in the JavaScript SDK.

Reply:
@tejasranade What do you think about using a symbol to separate the collection name and the tag? If we ever needed to separate the collection name and tag we could pass these as separate values lower down in the calls.

I can validate the tag against the regex we use for collection names. `^[a-zA-Z0-9-]*$`

Reply:
I was hoping to follow iOS lead on this, but looks like JS will be different.

What to think about - 
- Is it straightforward to identify a tagged collection in the cache vs. a regular one? Think about a debug scenario when someone tells us a table is messed up, for example.
- Can a tagged collection accidentally clash with a regular collection? For example, I use `Project` collection with tag `A`, and have another collection called `ProjectA` (terrible example, I know 😄 )

For lack of a better idea, how about something like `__`? I'm sure there's better options, that's the best I could quickly think of.
I'll leave it to you how to implement it i.e. how it's passed down. It doesn't need to be in the public API obv.

Reply:
Judging by the regex metioned in a previous comment, it seems like we can go with underscores - they are forbidden for backend collection names. One or more doesn't seem to matter since they are forbidden and can never be part of the actual collection's name. A consideration is collection name length, I guess, and support in different data storage providers (sqlite, websql, indexeddb, different platforms/implementations).

@heyzooi right now, tags create a separate collection in the same database, in JS, so things like empty strings should be rejected, because they would result in the same name (since there is no separator right now). Whitespaces can also be troublesome (tested in websql). For us, the matter is ensuring a subset of characters which are supported in all our storage providers (since we have many, some on more than one platform) and would be useful for customers.

I think alphanumeric characters make sense. Sounds to me like "_" might be part of someone's convention and might be desired for client use.

I can't think of a reason to not create a new database either - why not? Then again, I'm not too familiar with the scenario and what these scoped data stores would be used for outside of this client's use case. Cause we need to be thinking about the general use case, not just this particular one.